practical1/lib/tab1.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:practical1/models/user_bloc.dart';
+import 'loading.dart';
+import 'package:hive/hive.dart';
+import 'models/userdata.dart';
+import 'models/user_bloc.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+// class Tab1 extends StatefulWidget {
+//   @override
+//   _Tab1State createState() => _Tab1State();
+// }
+//
+// class _Tab1State extends State<Tab1> {
+//   void initState() {
+//     super.initState();
+//     Provider.of<AllUserData>(context, listen: false).getData();
+//   }
+//
+//   @override
+//   Widget build(BuildContext context) {
+//     return Container(
+//         child: (Provider.of<AllUserData>(context).allUsers.length != 0)
+//             ? DataList()
+//             : Loading());
+//   }
+// }
+
+class Tab1 extends StatefulWidget {
+  @override
+  _Tab1State createState() => _Tab1State();
+}
+
+class _Tab1State extends State<Tab1> {
+  ScrollController myScrollController = ScrollController();
+
+  void initState() {
+    super.initState();
+    BlocProvider.of<UserBloc>(context).add(GetData());
```

Comment:
1. When accessing provider in the initState, we must use `listen = false` to avoid exceptions
2. Rather here, we can add GetData() event while providing Bloc in parent widget itself